cogs/annual_handover_and_reset.py

```diff
@@ -0,0 +1,208 @@
+
+            logger.debug("Removing Committee-Elect role from user: %s", committee_elect_member)
+            await committee_elect_member.remove_roles(
+                committee_elect_role,
+                reason=f"{ctx.user} used TeX-Bot slash-command: \"handover\"",
+            )
+
+        await initial_response.edit(content=":white_check_mark: Handover procedure complete!")
+
+
+class AnnualResetRolesCommandCog(TeXBotBaseCog):
+    """Cog class that defines the reset_roles command."""
+
+    ACADEMIC_YEAR_ROLE_NAMES: Final[frozenset[str]] = frozenset(
+        {
+            "Foundation Year",
+            "First Year",
+            "Second Year",
+            "Final Year",
+            "Year In Industry",
+            "Year Abroad",
+            "PGT",
+        },
+    )
+
+    @discord.slash_command(  # type: ignore[no-untyped-call, misc]
+        name="annual_role_reset",
+        description="Removes the @Member role and academic year roles from all users",
+    )
+    @CommandChecks.check_interaction_user_has_committee_role
+    @CommandChecks.check_interaction_user_in_main_guild
+    async def annual_role_reset(self, ctx: TeXBotApplicationContext) -> None:
+        """
+        Definition & callback response of the "reset_roles" command.
+
+        The "reset_roles" command removes the "Member" and academic year roles
+        from any user that has them and subsequently deletes all instances of
+        the GroupMadeMember database model.
+        """
+        # NOTE: Shortcut accessors are placed at the top of the function, so that the exceptions they raise are displayed before any further errors may be sent
+        main_guild: discord.Guild = self.bot.main_guild
+        member_role: discord.Role = await self.bot.member_role
+
+        logger.debug("Reset roles command called.")
+        initial_response: discord.Interaction | discord.WebhookMessage = await ctx.respond(
+            ":hourglass: Resetting membership and year roles... :hourglass:",
+        )
+
+        member: discord.Member
+        for member in member_role.members:
+            await member.remove_roles(
+                member_role,
+                reason=f"{ctx.user} used TeX-Bot slash-command: \"/reset_roles\"",
+            )
+
+        logger.debug("Removed Member role from all users!")
+        await initial_response.edit(
+            content=":hourglass: Removed Member role from all users...",
+        )
+
+        await GroupMadeMember._default_manager.all().adelete()
+
+        await initial_response.edit(
+            content=":white_check_mark: Deleted all members from the database...",
+        )
+        logger.debug("Deleted all members from the database.")
+
+        year_roles: set[discord.Role] = {
+            role
+            for role_name
+            in self.ACADEMIC_YEAR_ROLE_NAMES
+            if isinstance(
+                (role := discord.utils.get(main_guild.roles, name=role_name)),
+                discord.Role,
+            )
```

Comment:
Unnecessary `isinstance` check. Use less expensive boolean existence check (this ensures the value is truthy and thus is not `None`)

Fix:
```python
year_roles: set[discord.Role] = {
    role
    for role_name
    in self.ACADEMIC_YEAR_ROLE_NAMES
    if (role := discord.utils.get(main_guild.roles, name=role_name))
}
```